go/vt/binlog/binlogplayer/binlog_player.go

```diff
@@ -466,59 +488,74 @@ func (blp *BinlogPlayer) ApplyBinlogEvents(ctx context.Context) error {
+  transaction_timestamp BIGINT(20) NOT NULL,
+  state VARBINARY(100) NOT NULL,
+  message VARBINARY(1000) DEFAULT NULL,
+  PRIMARY KEY (id)
+) ENGINE=InnoDB`}
+}
+
+// setVReplicationState updates the state in the _vt.vreplication table.
+func setVReplicationState(dbClient VtClient, uid uint32, state, message string) error {
+	query := fmt.Sprintf("UPDATE _vt.vreplication SET state='%v', message='%v' WHERE id=%v", state, message, uid)
+	if _, err := dbClient.ExecuteFetch(query, 1); err != nil {
+		return fmt.Errorf("could not set state: %v: %v", query, err)
+	}
+	return nil
 }
 
-// PopulateBlpCheckpoint returns a statement to populate the first value into
-// the _vt.blp_checkpoint table.
-func PopulateBlpCheckpoint(index uint32, position string, maxTPS int64, maxReplicationLag int64, timeUpdated int64, flags string) string {
-	return fmt.Sprintf("INSERT INTO _vt.blp_checkpoint "+
-		"(source_shard_uid, pos, max_tps, max_replication_lag, time_updated, transaction_timestamp, flags) "+
-		"VALUES (%v, '%v', %v, %v, %v, 0, '%v')",
-		index, position, maxTPS, maxReplicationLag, timeUpdated, flags)
+// CreateVReplication returns a statement to populate the first value into
+// the _vt.vreplication table.
+func CreateVReplication(index uint32, workflow string, source *binlogdatapb.BinlogSource, position string, maxTPS, maxReplicationLag, timeUpdated int64) string {
+	return fmt.Sprintf("INSERT INTO _vt.vreplication "+
+		"(id, workflow, source, pos, max_tps, max_replication_lag, time_updated, transaction_timestamp, state) "+
+		"VALUES (%v, '%v', '%s', '%v', %v, %v, %v, 0, '%v')",
+		index, workflow, source.String(), position, maxTPS, maxReplicationLag, timeUpdated, BlpStopped)
 }
 
-// updateBlpCheckpoint returns a statement to update a value in the
-// _vt.blp_checkpoint table.
-func updateBlpCheckpoint(uid uint32, pos mysql.Position, timeUpdated int64, txTimestamp int64) string {
+// updateVReplicationPos returns a statement to update a value in the
+// _vt.vreplication table.
+func updateVReplicationPos(uid uint32, pos mysql.Position, timeUpdated int64, txTimestamp int64) string {
 	if txTimestamp != 0 {
 		return fmt.Sprintf(
-			"UPDATE _vt.blp_checkpoint "+
+			"UPDATE _vt.vreplication "+
 				"SET pos='%v', time_updated=%v, transaction_timestamp=%v "+
-				"WHERE source_shard_uid=%v",
+				"WHERE id=%v",
 			mysql.EncodePosition(pos), timeUpdated, txTimestamp, uid)
 	}
 
 	return fmt.Sprintf(
-		"UPDATE _vt.blp_checkpoint "+
+		"UPDATE _vt.vreplication "+
 			"SET pos='%v', time_updated=%v "+
-			"WHERE source_shard_uid=%v",
+			"WHERE id=%v",
 		mysql.EncodePosition(pos), timeUpdated, uid)
 }
 
-// QueryBlpCheckpoint returns a statement to query the gtid and flags for a
-// given shard from the _vt.blp_checkpoint table.
-func QueryBlpCheckpoint(index uint32) string {
-	return fmt.Sprintf("SELECT pos, flags FROM _vt.blp_checkpoint WHERE source_shard_uid=%v", index)
+// ReadVReplicationPos returns a statement to query the gtid for a
```

Comment:
The term gtid made sense when this code was written because both MariaDB and MySQL overloaded the term, which was conveniently reused by us. But now, we're shoehorning other things in this. The ideal future term for this would be `VReplicationPosition`. The refactor is not worth it at this moment.